rewrite/codegen_simulator.rs

```diff
@@ -171,10 +170,12 @@ impl CodeGenSimulator {
     }
 
     fn get_maps(&mut self) {
-        for map in &mut self.ir.maps.clone() {
+        let mut maps = Vec::new();
+        mem::swap(&mut maps, &mut self.ir.maps);
```

Comment:
We should probably clean up this entire "get" process later. mem::swap should not be necessary here at all.

Reply:
Yeah if we're not mutating, we could probably use Arc. Also, we might want to move from vecs to hashes because we're calling `Vec::contains` which is kind of slow (unless n is pretty small?)